mache-chroniclemap/src/main/java/com/excelian/mache/chroniclemap/ChronicleMapMacheProvisioner.java

```diff
@@ -81,77 +74,386 @@ public ChronicleMapMacheProvisioner(int upperWaterMark, int lowerWaterMark, int
+        return this;
+    }
 
-        public ChronicleMapMacheBuilder withAcceptableWatermark(int acceptableWatermark) {
-            this.acceptableWatermark = acceptableWatermark;
-            return this;
-        }
+    /**
+     * Pass through to constantKeySizeBySample on ChronicleMapBuilder.
+     *
+     * @see ChronicleMapBuilder#constantKeySizeBySample(Object)
+     */
+    public ChronicleMapMacheProvisioner<K, V> constantKeySizeBySample(K sampleKey) {
+        configToApply.add((builder -> builder.constantKeySizeBySample(sampleKey)));
+        return this;
+    }
 
-        public ChronicleMapMacheBuilder withRunCleanupThread(boolean runCleanupThread) {
-            this.runCleanupThread = runCleanupThread;
-            return this;
-        }
+    /**
+     * Pass through to averageValueSize on ChronicleMapBuilder.
+     *
+     * @see ChronicleMapBuilder#averageValueSize(double)
+     */
+    public ChronicleMapMacheProvisioner<K, V> averageValueSize(double averageValueSize) {
+        configToApply.add((builder -> builder.averageValueSize(averageValueSize)));
+        return this;
+    }
 
-        public ChronicleMapMacheBuilder withRunNewThreadForCleanup(boolean runNewThreadForCleanup) {
-            this.runNewThreadForCleanup = runNewThreadForCleanup;
-            return this;
-        }
+    /**
+     * Pass through to constantValueSizeBySample on ChronicleMapBuilder.
+     *
+     * @see ChronicleMapBuilder#constantValueSizeBySample(Object)
+     */
+    public ChronicleMapMacheProvisioner<K, V> constantValueSizeBySample(V sampleValue) {
+        configToApply.add((builder -> builder.constantValueSizeBySample(sampleValue)));
+        return this;
+    }
 
-        public ChronicleMapMacheBuilder withEvictionListener(
-                ConcurrentLRUCache.EvictionListener<K, V> evictionListener) {
-            this.evictionListener = evictionListener;
-            return this;
-        }
+    /**
+     * Pass through to actualChunkSize on ChronicleMapBuilder.
+     *
+     * @see ChronicleMapBuilder#actualChunkSize(int)
+     */
+    public ChronicleMapMacheProvisioner<K, V> actualChunkSize(int actualChunkSize) {
+        configToApply.add((builder -> builder.actualChunkSize(actualChunkSize)));
+        return this;
+    }
 
-        public ChronicleMapMacheProvisioner<K, V> create() {
-            return new ChronicleMapMacheProvisioner<>(upperWaterMark, lowerWaterMark, acceptableWatermark,
-                    runCleanupThread, runNewThreadForCleanup, evictionListener, chronicleMapBuilder, persistedTo);
-        }
+    /**
+     * Pass through to maxChunksPerEntry on ChronicleMapBuilder.
+     *
+     * @see ChronicleMapBuilder#maxChunksPerEntry(int)
+     */
+    public ChronicleMapMacheProvisioner<K, V> maxChunksPerEntry(int maxChunksPerEntry) {
+        configToApply.add((builder -> builder.maxChunksPerEntry(maxChunksPerEntry)));
+        return this;
+    }
```

Comment:
The thing I don't like about using the same instance as the builder (i.e. `return this;`), it gives no grammar for what next steps are optional Vs. steps that are mandatory...the downside of the way I've done it elsewhere is lots of interfaces, and the code starts to resemble spaghetti.  I think when we next have the opportunity, we should weigh up the relative merits of both approaches and standardise on one.
This is not worth halting a PR for while this is still up for debate.